ldap-client.go

```diff
@@ -254,6 +256,50 @@ func (lc *LDAPClient) GetAllGroupsByName(groupName string) ([]LdapGroup, error)
 	return groups, nil
 }
 
+// GetAllGroupsWithMembersByName returns list of groups matching a name.
+func (lc *LDAPClient) GetAllGroupsWithMembersByName(groupsDN string, groupCN []string) ([]*LdapGroup, error) {
+	err := lc.Connect()
+	if err != nil {
+		return nil, err
+	}
+
+	dn := lc.BindDN
+	if groupsDN != "" {
+		dn = groupsDN
+	}
+	// First bind with a read only user
+	if dn != "" && lc.BindPassword != "" {
+		if err = lc.Conn.Bind(dn, lc.BindPassword); err != nil {
+			return nil, nil
+		}
+	}
+
+	searchFilter := fmt.Sprintf("(|%s)", strings.Join(groupCN, ""))
+
+	searchRequest := ldap.NewSearchRequest(
+		lc.Base,
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
```

Comment:
If I remember correctly, the NeverDerefAliases says we will get the alias entry without the actual entry he aliases to.
Was it tested with aliases groups?

Reply:
As we talked. It needs to be further discussed with the Product just to make sure, but it's out of the scope of reviewing the code 🙏🏻